block_darwin.go

```diff
@@ -0,0 +1,250 @@
+	MountPoint                                  string // /
+	ParentWholeDisk                             string // disk1
+	PartitionMapPartition                       bool   // false
+	RAIDMaster                                  bool   // false
+	RAIDSlice                                   bool   // false
+	RecoveryDeviceIdentifier                    string // disk1s3
+	Removable                                   bool   // false
+	RemovableMedia                              bool   // false
+	RemovableMediaOrExternalDevice              bool   // false
+	SMARTStatus                                 string // Verified
+	Size                                        int64  // 499963174912
+	SolidState                                  bool   // true
+	SupportsGlobalPermissionsDisable            bool   // true
+	SystemImage                                 bool   // false
+	TotalSize                                   int64  // 499963174912
+	VolumeAllocationBlockSize                   int64  // 4096
+	VolumeName                                  string // Macintosh HD
+	VolumeSize                                  int64  // 499963174912
+	VolumeUUID                                  string // some-uuid-foo-bar
+	WholeDisk                                   bool   // false
+	Writable                                    bool   // true
+	WritableMedia                               bool   // true
+	WritableVolume                              bool   // true
+	// also has a SMARTDeviceSpecificKeysMayVaryNotGuaranteed dict with various info
+	// NOTE: VolumeUUID sometimes == DiskUUID, but not always. So far Content is always a different UUID.
+}
+
+type ioregPlist struct {
+	// there's a lot more than just this...
+	ModelNumber  string `plist:"Model Number"`
+	SerialNumber string `plist:"Serial Number"`
+	VendorName   string `plist:"Vendor Name"`
+}
+
+func (ctx *context) getDiskUtilListPlist() (diskUtilListPlist, error) {
+	out, err := exec.Command("diskutil", "list", "-plist").Output()
+	if err != nil {
+		return diskUtilListPlist{}, errors.Wrap(err, "diskutil list failed")
+	}
+
+	var data diskUtilListPlist
+	if _, err := plist.Unmarshal(out, &data); err != nil {
+		return diskUtilListPlist{}, errors.Wrap(err, "diskutil list plist unmarshal failed")
+	}
+
+	return data, nil
+}
+
+func (ctx *context) getDiskUtilInfoPlist(device string) (diskUtilInfoPlist, error) {
+	out, err := exec.Command("diskutil", "info", "-plist", device).Output()
+	if err != nil {
+		return diskUtilInfoPlist{}, errors.Wrapf(err, "diskutil info for %q failed", device)
+	}
+
+	var data diskUtilInfoPlist
+	if _, err := plist.Unmarshal(out, &data); err != nil {
+		return diskUtilInfoPlist{}, errors.Wrapf(err, "diskutil info plist unmarshal for %q failed", device)
+	}
+
+	return data, nil
+}
+
+func (ctx *context) getIoregPlist(ioDeviceTreePath string) (ioregPlist, error) {
+	elems := strings.Split(ioDeviceTreePath, "/")
+	name := elems[len(elems)-1]
```

Comment:
For the above, please use:

```
name := filepath.Base(ioDeviceTreePath)
```

Reply:
Fixed, except using path.Base instead of filepath.Base, since the iokit device tree is its own thing and has nothing to do with the filesystem.